coalib/__init__.py

```diff
@@ -6,10 +6,61 @@
 
 
 import sys
+
+from os import chdir, getcwd
 from os.path import join, dirname
+from subprocess import check_output, CalledProcessError
+
+
+COALIB_DIR = join(dirname(__file__))
+
+VERSION_FILE = join(COALIB_DIR, 'VERSION')
+
+COMMIT_FILE = join(COALIB_DIR, 'COMMIT')
+
 
+def get_current_commit(git_dir):
+    '''
+    :param git_dir:     string containing the directory to check the
+                        current commit in.
+    :return:            string containing the current commit sha if `git_dir`
+                        is initialized as a git repo otherwise an empty string.
+    '''
+    old_dir = getcwd()
+    chdir(git_dir)
 
-VERSION_FILE = join(dirname(__file__), 'VERSION')
+    try:
+        commit = check_output(['git', 'rev-parse', 'HEAD'])
+        commit = commit.decode('utf-8').strip()
+
+        status = check_output(['git', 'status', '--porcelain'])
+        status = status.decode('utf-8').strip()
+        if status != '':
+            commit += '*'
+    except (OSError, CalledProcessError) as e:
+        commit = ''
+
+    chdir(old_dir)
+    return commit
```

Comment:
Maybe you could use `change_directory` from `coala_utils` here to shorten this up  a bit.